### PR TITLE
[SMTChecker] Allow SymbolicVariable from smt::Sort

### PR DESCRIPTION
**Pull request by leonardoalt**

### CODE REVIEW

File: libsolidity/formal/SymbolicVariables.h [hunk 1]
```diff
@@ -40,6 +40,11 @@ class SymbolicVariable
 		std::string const& _uniqueName,
 		smt::SolverInterface& _interface
 	);
+	SymbolicVariable(
+		smt::SortPointer _sort,
+		std::string const& _uniqueName,
```
Comment (ekpyron) [inline libsolidity/formal/SymbolicVariables.h]: Just as a side-remark: In both constructors here you should actually take the ``string`` by value and move it into the member. Can happen in a separate PR, though.

**Review by ekpyron — DISMISSED** [review]

**Review by erak — DISMISSED** [review]

File: libsolidity/formal/SymbolicVariables.h (hunk 1 above)
Comment (leonardoalt) [reply libsolidity/formal/SymbolicVariables.h]: I'll do it on this one, just a sec

Comment (ekpyron) [reply libsolidity/formal/SymbolicVariables.h]: If you do it, I'd do it for both at once ;-).

Comment (leonardoalt) [reply libsolidity/formal/SymbolicVariables.h]: Updated

**Review by ekpyron — DISMISSED** [review]

File: libsolidity/formal/SymbolicVariables.h (hunk 1 above)
Comment (leonardoalt) [reply libsolidity/formal/SymbolicVariables.h]: Not only both constructors, but all the ones of the derived classes ;)

File: libsolidity/formal/SymbolicVariables.cpp [hunk 2]
```diff
@@ -26,14 +26,30 @@ using namespace dev::solidity;
 
 SymbolicVariable::SymbolicVariable(
```
Comment (chriseth) [inline libsolidity/formal/SymbolicVariables.cpp:27]: It would be nice to call the other constructor, but then you cannot `solAssert(m_type`, can you?

**Review by chriseth — APPROVED** [review]

**Pull request merged**